rcl/src/rcl/security_directory.c

```diff
@@ -219,13 +219,15 @@ const char * rcl_get_secure_root(
   char * lookup_strategy = NULL;
   char * node_secure_root = NULL;
   if (ros_secure_node_override) {
-    node_secure_root = ros_secure_root_env;
+    node_secure_root = (char *)allocator->allocate(ros_secure_root_size + 1, allocator->state);
+    strcpy(node_secure_root, ros_secure_root_env);
```

Comment:
strncpy? as we have the legnth of `ros_secure_root_size`. `strncpy` may have issues on Windows IIRC so we may want to be careful there.

Reply:
thanks for catching this! used `mempy` instead.